ext/neural_snow/ModelTools.jl

```diff
@@ -244,6 +247,94 @@ function make_timeseries(
+z and SWE model. Used in the paper for combined timeseries.
+
+# Arguments
+- `zmodel`: The model used for depth forecasting (can be any model with a defined `evaluate` call).
+- `swemodel`: The model used for SWE forecasting (can be any model with a defined `evaluate` call).
+- `timeseries::DataFrame`: The input data frame used to generate predictions, including a time variable.
+- `dt::Period`: The unit timestep present in the dataframe (i.e. daily dataframe, `dt = Day(1)` or `Second(86400)`).
+- `timevar::Symbol`: The variable giving the time of each forcing. Default is `:date`.
+- `z_inputvars::Vector{Symbol}`: The variables (in order), to extract from the data to use for depth predictions.
+Default is `[:z, :SWE, :rel_hum_avg, :sol_rad_avg, :wind_speed_avg, :air_temp_avg, :dprecipdt_snow]` like the paper.
+- `SWE_inputvars::Vector{Symbol}`: The variables (in order), to extract from the data to use for SWE predictions.
+Default is `[:z, :SWE, :rel_hum_avg, :sol_rad_avg, :wind_speed_avg, :air_temp_avg, :dprecipdt_snow]` like the paper.
+- `dtype::Type`: The data type required for input to the model. Default is `Float32`.
+- `hole_thresh::Int`: The acceptable number of "holes" in the timeseries for the model to skip over. Default is 30.
+"""
+function paired_timeseries(
+    zmodel,
+    swemodel,
+    timeseries::DataFrame,
+    dt::Period;
+    timevar::Symbol = :date,
+    z_inputvars::Vector{Symbol} = [
+        :z,
+        :SWE,
+        :rel_hum_avg,
+        :sol_rad_avg,
+        :wind_speed_avg,
+        :air_temp_avg,
+        :dprecipdt_snow,
+    ],
+    SWE_inputvars::Vector{Symbol} = [
+        :z,
+        :SWE,
+        :rel_hum_avg,
+        :sol_rad_avg,
+        :wind_speed_avg,
+        :air_temp_avg,
+        :dprecipdt_snow,
+    ],
+    dtype::Type = Float32,
+    hole_thresh::Int = 30,
+)
+    zforcings = Matrix{dtype}(select(timeseries, z_inputvars))
+    sweforcings = Matrix{dtype}(select(timeseries, SWE_inputvars))
+    z_zidx = findfirst(z_inputvars .== :z)
+    z_sweidx = findfirst(z_inputvars .== :SWE)
+    swe_zidx = findfirst(SWE_inputvars .== :z)
+    swe_sweidx = findfirst(SWE_inputvars .== :SWE)
+    check_dates =
+        (timeseries[2:end, timevar] - timeseries[1:(end - 1), timevar]) ./ dt
+    pred_dzs = zeros(nrow(timeseries) - 1)
+    pred_dswes = zeros(nrow(timeseries) - 1)
+    pred_zs = zeros(nrow(timeseries))
+    pred_swes = zeros(nrow(timeseries))
+    pred_zs[1] = timeseries[1, :z]
+    pred_swes[1] = timeseries[1, :SWE]
+    countresets = 0
+    for j in 2:length(pred_zs)
+        zinput = zforcings[j - 1, :]
+        sweinput = sweforcings[j - 1, :]
+        zinput[z_zidx] = pred_zs[j - 1]
+        zinput[z_sweidx] = pred_swes[j - 1]
+        sweinput[swe_zidx] = pred_zs[j - 1]
+        sweinput[swe_sweidx] = pred_swes[j - 1]
+        zpred = evaluate(zmodel, zinput)[1]
+        swepred = evaluate(swemodel, sweinput)[1]
+        pred_dzs[j - 1] = zpred
```

Comment:
It might be helpful to use more descriptive/consistent names - is the output of evaluate a tendency/time derivative, or the next value?  Here you say something like "dz = z" 
`pred_dzs[j - 1] = zpred`

below you say `new_z = pred_z + dt*zpred`

Reply:
I left "zpred" as indicative of "the prediction made by the z-model as opposed to the swe-model", though I agree this could be ambiguous and instead interpreted as "prediction of depth z". If unit-based-names are more desired, the name would be "dzdt_pred" and pred_dzs would be better written as "pred_dzdts"